### DISCUSSION TITLE
User defined interface

### DISCUSSION THREAD

**Discussion by ashishdhole** [idaholab/moose#27815] [Q&A Modules: Phase field]
Hello, 
Can we define the interface in the form of a function. For example. Let the simulation box contain 2 grains but instead of them separated by a straight interface, can I define say interface as a sin function (wavy type)? 
Thank you
Ashish

**Comment by GiudGiud** [comment]
hello

The interface is defined with a variable right?
You could use a Function and a FunctionIC to initialize said variable

Guillaume

**Reply by ashishdhole** [reply]
Say for example, I use SmoothCircleIC, the function is a circle with invalue =1 and outvalue = 0. but if I define a sine function, it will be difficult to decide which side of the function is 1 and which side is 0, if eta is my variable on which AC kernels are acting.

**Reply by GiudGiud** [reply]
You can set a threshold. FunctionIC can use a ParsedFunction for example, in which you can do:

for example this could be an `expression` that draws thresholds from a sin function
```
if(sin(x)>0, this value, if(sin(x)<-0.5, this other value, another value)
```
